kasa/iotprotocol.py

```diff
@@ -85,16 +86,16 @@ async def _query(self, request: str, retry_count: int = 3) -> Dict:
         raise SmartDeviceException("Query reached somehow to unreachable")
 
     async def _execute_query(self, request: str, retry_count: int) -> Dict:
-        if self._transport.needs_handshake:
-            await self._transport.handshake()
+        if self.transport.needs_handshake:
```

Comment:
I don't think we should be making these public as they are clearly internal.

Reply:
I've tended to think of the underscore as internal to the class rather than the library.  I don't have an issue with using underscore for internal with the library but as per comment above I'll need to update the base classes to have the underscore as well. Apologies for that.

Reply:
No worries :-) So using `_` is merely a convention to indicate that those are internals that are not part of the external API. This is a good practice as it delivers the intention to anyone using the lib / reading the code.

There is also a `__` which is more than a convention (i.e., it mangles the names) but that's rather seldomly useful. Here's a rather clean summary: https://stackoverflow.com/a/27481645/1368594

Reply:
Done